src/Notes/NoteTraits.php

```diff
@@ -33,4 +33,37 @@ public static function wc_admin_active_for( $seconds ) {
 
 		return ( ( time() - $wc_admin_installed ) >= $seconds );
 	}
+
+	/**
+	 * Check if the note has been previously added.
+	 */
+	public static function note_exists() {
+		$data_store = \WC_Data_Store::load( 'admin-note' );
+		$note_ids   = $data_store->get_notes_with_name( self::NOTE_NAME );
+		return ! empty( $note_ids );
+	}
+
+	/**
+	 * Add the note if it passes predefined conditions.
+	 */
+	public static function possibly_add_note() {
+		if ( self::note_exists() ) {
+			return;
+		}
+
+		$note = self::get_note();
+
+		if ( ! $note instanceof WC_Admin_Note ) {
+			return;
+		}
+
+		if (
+			'no' === get_option( 'woocommerce_show_marketplace_suggestions', 'yes' ) &&
+			WC_Admin_Note::E_WC_ADMIN_NOTE_MARKETING === $note->get_type()
```

Comment:
I think it would be great to also check this at the `save()` level to prevent bypassing this, but currently there does not appear to be a way of doing this.

Filtering allowed types is one option, but this throws an exception and would require a bigger refactor.